src/wasm/wasm-type.cpp

```diff
@@ -43,49 +43,46 @@ size_t hash<wasm::Type>::operator()(const wasm::Type& type) const {
 }
 
 size_t hash<wasm::Signature>::operator()(const wasm::Signature& sig) const {
-  auto res = hash<uint64_t>{}(sig.params.getID());
-  wasm::hash_combine(res, sig.results.getID());
-  return res;
+  return wasm::rehash(uint64_t(hash<uint64_t>{}(sig.params.getID())),
+                      uint64_t(hash<uint64_t>{}(sig.results.getID())));
 }
 
 size_t hash<wasm::Field>::operator()(const wasm::Field& field) const {
-  auto res = hash<uint64_t>{}(field.type.getID());
-  wasm::hash_combine(res, uint32_t(field.packedType));
-  wasm::hash_combine(res, field.mutable_);
-  return res;
+  return wasm::rehash(uint64_t(hash<uint64_t>{}(field.type.getID())),
+                      uint64_t(field.packedType) << 1 | field.mutable_);
```

Comment:
This overload is new and does what the previous old implementation does for `getID()`. If my assumption that piping through `std::hash` is bad is correct, than this should better not pipe, yet it will still return the final `uint64_t` hash value as `size_t`.

Reply:
Also, since `getID()` is `uint64_t` and may be implicitly converted from the `uintptr_t` address of a constructed type, it seems that making this deterministic between archs isn't easy in the first place.

Reply:
If we need to support deterministic hashing of `Type`, I think we would have to hash the structure of the `Type` rather than its ID. That seems like it would be fine.

Reply:
Good point!